qu19041_ex3.py

```diff
@@ -0,0 +1,76 @@
+import numpy as np
+import matplotlib.pyplot as pyplot
+
+# PART (A)
+
+def integrand(screen_coord, apt_coord, c):
+    return np.exp( c['waveno'] * 1j * (screen_coord - apt_coord) ** 2 / (2 * c['screendist']) )
+
+def fresnel_integral(screen_coord, apt, c):
+    assert c['intervals'] % 2 == 0, "number of intervals must be even."
+
+    aperture = np.linspace(apt['apt_lower'], apt['apt_upper'], c['intervals'] + 1)
+    d_aperture = aperture[1] - aperture[0]
+    sum_vals = np.zeros( len(aperture), dtype=np.complex_ )
+    
+    for i in range( len(aperture) ):
+        sum_vals[i] = integrand(screen_coord, aperture[i], c)
+
+    return c['multiplier'] * d_aperture/3 * \
+    ( sum_vals[0] + np.sum(sum_vals[0:-1:2] + 4*sum_vals[1::2] + 2*sum_vals[2::2]))
```

Comment:
Yo, spaces around your operators :)

Reply:
Can't you indent this? Just to make it clearer that it belongs to the return statement. Small thing, just a bit clearer.